src/fsharp/Interactive.DependencyManager/DependencyManager.fs

```diff
@@ -0,0 +1,246 @@
+            yield Path.GetDirectoryName assemblyLocation
+            yield AppDomain.CurrentDomain.BaseDirectory
+        ])
+
+    let enumerateDependencyManagerAssemblies compilerTools reportError =
+        getCompilerToolsDesignTimeAssemblyPaths compilerTools
+        |> Seq.append (assemblySearchPaths.Force())
+        |> Seq.collect (fun path ->
+            try
+                if Directory.Exists(path) then Directory.EnumerateFiles(path, dependencyManagerPattern)
+                else Seq.empty
+            with _ -> Seq.empty)
+        |> Seq.choose (fun path -> 
+            try
+                Some(Assembly.LoadFrom path)
+            with
+            | e ->
+                let e = stripTieWrapper e
+                reportError ErrorReportType.Warning (InteractiveDependencyManager.SR.couldNotLoadDependencyManagerExtension(path,e.Message))
+                None)
+        |> Seq.filter (fun a -> assemblyHasAttribute a dependencyManagerAttributeName)
+
+    let mutable registeredDependencyManagers: Map<string, IDependencyManagerProvider> option= None
+
+    let RegisteredDependencyManagers (compilerTools: string seq) (outputDir: string option) (reportError: ErrorReportType -> int * string -> unit) =
+        match registeredDependencyManagers with
+        | Some managers ->
+            managers
+        | None ->
+            let managers =
+                let defaultProviders =[]
+
+                let loadedProviders =
+                    enumerateDependencyManagerAssemblies compilerTools reportError
+                    |> Seq.collect (fun a -> a.GetTypes())
+                    |> Seq.choose (fun t -> ReflectionDependencyManagerProvider.InstanceMaker(t, outputDir))
+                    |> Seq.map (fun maker -> maker ())
+
+                defaultProviders
+                |> Seq.append loadedProviders
+                |> Seq.map (fun pm -> pm.Key, pm)
+                |> Map.ofSeq
+
+            registeredDependencyManagers <-
+                if managers.Count > 0 then
+                    Some managers
+                else
+                    None
+            managers
+
+    member _.CreatePackageManagerUnknownError(compilerTools: string seq, outputDir: string, packageManagerKey: string, reportError: ErrorReportType -> int * string -> unit) =
+
+        let registeredKeys = String.Join(", ", RegisteredDependencyManagers compilerTools (Option.ofString outputDir) reportError |> Seq.map (fun kv -> kv.Value.Key))
+        let searchPaths = assemblySearchPaths.Force()
+        InteractiveDependencyManager.SR.packageManagerUnknown(packageManagerKey, String.Join(", ", searchPaths, compilerTools), registeredKeys)
+
+    member this.TryFindDependencyManagerInPath (compilerTools: string seq, outputDir: string, reportError: ErrorReportType -> int * string -> unit, path: string): string * IDependencyManagerProvider =
+        try
+            if path.Contains ":" && not (System.IO.Path.IsPathRooted path) then
```

Comment:
`System.IO` is already open so no need to fully quality the name here

Reply:
agreed.